### PR TITLE
add SmoothExtractor

### PR DESCRIPTION
**Pull request by sunya-ch**
According to feature request issue in https://github.com/sustainable-computing-io/kepler-model-server/issues/122, this PR implements SmoothExtractor to smooth the data with specific window size on extraction. 
This is also available to test via the entrypoint `cmd/main.py` with `--extractor smooth`.

As this PR has a base on https://github.com/sustainable-computing-io/kepler-model-server/pull/121 fix, please merge the fix first.

Signed-off-by: Sunyanan Choochotkaew <sunyanan.choochotkaew1@ibm.com>

### CODE REVIEW

File: src/train/trainer/__init__.py [hunk 1]
```diff
@@ -233,7 +236,7 @@ def save_model_and_metadata(self, node_type, X_test, y_test):
         mae_map = dict()
         item = self.get_basic_metadata(node_type)
         for component in self.energy_components:
-            mae = self.get_mae(node_type, component, X_test, y_test)
+            mae = self.get_mae(node_type, component, X_test_map[component], y_test_map[component])
```
Comment (rootfs) [inline src/train/trainer/__init__.py:239]: is this still needed after #121?

**Comment by rootfs** [comment]
does kepler also need to smooth the feature during inference?

**Comment by sunya-ch** [comment]
> does kepler also need to smooth the feature during inference?

I don't think so because we return one-to-one prediction without historical data for the current power model.

File: src/train/trainer/__init__.py (hunk 1 above)
Comment (sunya-ch) [reply src/train/trainer/__init__.py:239]: I will rebase the PR.

Comment (sunya-ch) [reply src/train/trainer/__init__.py]: Done rebasing. Please confirm it. Thanks.

**Review by rootfs — APPROVED** [review]

**Pull request merged**